src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -255,6 +267,47 @@ private void readRcaEnabledFromConf() {
         });
   }
 
+  /**
+   * Reads the mutedRCAList value from the rca.conf file, performs validation on the param value
+   * provided and on successful validation, updates the AnalysisGraph with muted RCA value.
+   *
+   * <p>In case all the RCAs in param value are incorrect, return without any update.
+   */
+  private void readAndUpdateMutesRcas() {
+    // If the rca config file has been updated since the lastModifiedTimeInMillisInMemory in memory,
+    // refresh the `muted-rcas` value from rca config file.
+    final RcaConf rcaConf = RcaControllerHelper.pickRcaConfForRole(currentRole);
+    long lastModifiedTimeInMillisOnDisk = rcaConf.getLastModifiedTime();
+    if (lastModifiedTimeInMillisOnDisk > lastModifiedTimeInMillisInMemory) {
+      try {
+        List<String> rcasForMute = rcaConf.getMutedRcaList();
+        LOG.info("RCAs provided for muting : {}", rcasForMute);
+
+        Set<String> graphNodeNames = new HashSet<>();
+        RcaUtil.getAnalysisGraphComponents(rcaConf).forEach(
+                connectedComponent -> graphNodeNames.addAll(connectedComponent.getNodeNames()));
+
+        Set<String> validRcasForMute = rcasForMute.stream()
+                .filter(rcaForMute -> graphNodeNames.contains(rcaForMute))
+                .collect(Collectors.toSet());
+
+        // If validRcasForMute is empty but rcasForMute is not empty
+        // all the input RCAs are incorrect, return.
+        if (validRcasForMute.isEmpty() && !rcasForMute.isEmpty()) {
+          LOG.info("Incorrect RCA value, cannot be muted : {}", rcasForMute);
```

Comment:
`error` log ?

Reply:
Thinking more about it, when people make the API call from the writer, they wouldn't even know as part of the API that they are referring to a non-existent RCA or other error. It will be silently emitted in the logs but the API will return OK status.

Because the Reader has the knowledge of the list of RCAs, it might make sense for the API on the writer to just proxy this to the reader and the READER do a validation and return error as the response itself and not just as silent logs. Thoughts ?

Reply:
We can discuss this further since this behavior would involve more elaborate changes through the code. Currently, the expectation is for the operator to either perform a GET after PUT to verify the muted API result or check the logs for errors. 

I agree this is something we should ideally do.